shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1380,6 +1411,31 @@ - (void)startKeyBoardAnimation:(NSTimeInterval)duration {
       }];
 }
 
+- (void)setupKeyboardAnimationCurveIfNeeded:(CAAnimation*)keyboardAnimation
+                                   duration:(double)duration {
+  // Set keyboard spring animation details (if animation is CASpringAnimation).
```

Comment:
Seems that we don't need this `duration` param ,also don't need `self.keyboardAnimationDuration`
Because the algorithm calculate instantly is calculate using relative time.

Reply:
Oops, missed that - but does make me think we could keep it and use `duration * 2` for the spring's `settlingDuration` rather than hardcoding it at .5. Don't think we can know if doubling the given duration will hold until Apple changes the spring curve details and we can test it 😬

Reply:
haha I mean, we don't need to now the duration of this animation.^_^ The algorithm will calculate the interpolate instantly using the start time and target time. So the duration is uselessO(∩_∩)O

Reply:
Haha guess that's what late night brain get's me! Glad we don't need the duration anymore, will remove tomorrow